forest/src/daemon.rs

```diff
@@ -398,6 +392,35 @@ pub(super) async fn start(config: Config, detached: bool) {
     info!("Forest finish shutdown");
 }
 
+async fn prompt_and_fetch_snapshot(config: &mut Config) {
```

Comment:
You can use this crate [atty](https://crates.io/crates/atty) to test if stdin is a tty and if it's the case ask the user for download of a snapshot.